src/main/java/org/kohsuke/groovy/sandbox/SandboxTransformer.java

```diff
@@ -704,7 +704,7 @@ public void visitExpressionStatement(ExpressionStatement es) {
                     if (mightBePositionalArgumentConstructor((VariableExpression) leftExpression)) {
                         CastExpression ce = new CastExpression(leftExpression.getType(), de.getRightExpression());
                         ce.setCoerce(true);
-                        es.setExpression(transform(new DeclarationExpression(leftExpression, de.getOperation(), transform(ce))));
+                        es.setExpression(transform(new DeclarationExpression(leftExpression, de.getOperation(), ce)));
```

Comment:
Well that is probably too low level. You care about the results, not the mechanism. I am just asking whether there is some automated test proving that the RHS is checked in the sandbox after this fix.